src/java.base/share/classes/com/sun/crypto/provider/Poly1305.java

```diff
@@ -255,6 +254,23 @@ private void processMultipleBlocks(byte[] input, int offset, int length, long[]
         }
     }
 
+    private void processMultipleBlocks(ByteBuffer buf, int blockMultipleLength) {
+        if (buf.hasArray()) {
+            byte[] input = buf.array();
+            int offset = buf.arrayOffset();
```

Comment:
arrayOffset returns the offset of the first element of the backing buffer. Need to also take into account the position() of the buffer here. We should also add test case for this.

Reply:
Thanks for catching this!! Fixed the mainline to handle position(). I have now reworked the test case quite a bit.